File: Templates/Challenge/Hello_World/Compute/template.pl
#!/opt/perl/bin/perl

use 5.032;

use strict;
use warnings;
no  warnings 'syntax';

use experimental 'signatures';
use experimental 'lexical_subs';

#
# See ../README.md
#

#
# Run as: perl ch-1.pl [plain|fetch|compute]
#

my $TYPE_PLAIN   =  0;
my $TYPE_FETCH   =  1;
my $TYPE_COMPUTE =  2;

my $type = $TYPE_PLAIN;   # Default
   $type = $TYPE_FETCH   if @ARGV && $ARGV [0] eq "fetch";
   $type = $TYPE_COMPUTE if @ARGV && $ARGV [0] eq "compute";


if ($type == $TYPE_PLAIN) {
    say "%TEXT";
}
elsif ($type == $TYPE_FETCH) {
    use OEIS;
    $, = ", ";
    say oeis (XXX => 0);
}
elsif ($type == $TYPE_COMPUTE) {
}

File: Templates/Challenge/Hello_World/Compute/template.c
# include <stdlib.h>
# include <stdio.h>
# include <string.h>

/*
 * See ../README.md
 */

/*
 * Run as: cc -o ch-1.o ch-1.c; ./ch-1.o [plain | compute]
 */

# define PLAIN    0
# define COMPUTE  1

typedef int number;  /* Change if we want large numbers */
int main (int argc, char * argv []) {
    int type = PLAIN;
    if (argc > 1) {
        if (strncmp (argv [1], "compute", 8) == 0) {
            type = COMPUTE;
        }
    }

    if (type == PLAIN) {
        printf ("%TEXT\n");
    }
    if (type == COMPUTE) {
    }
    exit (0);
}
